src/selectors/columnWidths.js

```diff
@@ -47,10 +48,11 @@ function columnWidths(columnGroupProps, columnProps, scrollEnabledY, width) {
     newColumnGroupProps,
     newColumnProps,
   } = flexWidths(columnGroupProps, columnProps, viewportWidth);
-  const [
+  const {
     fixedColumns,
+    fixedRightColumns,
     scrollableColumns,
-  ] = partition(newColumnProps, column => column.fixed);
+  } = groupBy(newColumnProps, getColumnCategory);
```

Comment:
Is this fine? I changed to `groupBy`since we need 3 groups here (partition only leaves 2).

Reply:
My bad, this logic is wrong. 
If columns in a category doesn't exist, then the property will be undefined. We need an empty array in this case.
Will change.